src/main/java/org/icgc/argo/workflow_management/service/WorkflowStatusMonitor.java

```diff
@@ -0,0 +1,145 @@
+package org.icgc.argo.workflow_management.service;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.icgc.argo.workflow_management.service.model.Metadata;
+import org.icgc.argo.workflow_management.service.model.Workflow;
+import org.icgc.argo.workflow_management.service.model.WorkflowErrorEvent;
+import org.icgc.argo.workflow_management.service.properties.NextflowProperties;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOError;
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.lang.String.format;
+import static java.time.OffsetDateTime.now;
+import static java.time.OffsetDateTime.parse;
+import static org.icgc.argo.workflow_management.util.JsonUtils.toJsonString;
+
+@Slf4j
+public class WorkflowStatusMonitor implements Runnable {
```

Comment:
This is very nextflow specific, either make a more generic monitor and implement a nextflow sub or make this explicitly nextflow only